### PR TITLE
`jj resolve --list`: Add descriptions of conflict complexity

### PR DESCRIPTION
**Pull request by ilyagr**
The descriptions focus on adds, not the deletions. The deletions are used
to compute the number of deleted files that must be sides of the conflict.

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- (not planned) I have updated the documentation (README.md, docs/, demos/)
- [x] I have added tests to cover my changes

------------------------------------

This is based on top of #975. I am pretty happy with the English of the descriptions. It might be nice to have prettier alignment, though.

There could also be a `--short` option that says the same thing more compactly, e.g. "txt, txt" or "dir, exe" for 2-sided conflicts.

On the other hand, I don't run into conflicts this complex often enough to know what is useful here.

### CODE REVIEW

**Comment by martinvonz** [comment]
> This is based on top of https://github.com/martinvonz/jj/pull/975

You can set this PR's "base" branch to ig/resolve_list to reflect that. Then GitHub will show only the additional commit in the list of commits.

**Comment by ilyagr** [comment]
> > This is based on top of #975
> 
> You can set this PR's "base" branch to ig/resolve_list to reflect that. Then GitHub will show only the additional commit in the list of commits.

Thanks! I guess I never noticed it since this only works for projects that are willing to make me a contributor (I think). I wonder if it will be clever enough to rebase it on main after the parent PR gets merged.

File: src/commands.rs [hunk 1]
```diff
@@ -2435,20 +2437,54 @@ fn cmd_resolve(
...
-        // E.g. we could have a letter per add : `FF` is a usual conflict
-        // between two versions of a file, `FD` is a file vs directory,
-        // `FFF` for a merge of three conflicting versions. Additionally,
-        // if (# removes) + 1 > (# adds), this indicates the file was deleted
-        // in some versions of the conflict. Perhaps that should be `R` for removed.
+    for (repo_path, conflict_id) in conflicts.iter() {
+        let conflict = tree.store().read_conflict(repo_path, conflict_id)?;
+        let sides = conflict.adds.len();
+        let deletions: i32 =
+            1 + i32::try_from(conflict.removes.len()).unwrap() - i32::try_from(sides).unwrap();
```
Comment (ilyagr) [inline src/commands.rs:2448]: Is there a better way to do arithmetic?

Comment (ilyagr) [reply src/commands.rs:2448]: Actually, I was over-complicating things. No need for arithmetic that complicated.

**Comment by martinvonz** [comment]
The first commit (the rename) looks empty

**Review by martinvonz — APPROVED** [review]

File: tests/test_resolve_command.rs [hunk 2]
```diff
@@ -465,6 +474,56 @@ fn test_file_vs_dir() {
...
 }
 
+#[test]
+fn test_description_with_dir_and_deletion() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "base", &[], &[("file", "base\n")]);
+    create_commit(&test_env, &repo_path, "sane", &["base"], &[("file", "b\n")]);
```
[the remark this answers is not in the document]
Comment (ilyagr) [reply tests/test_resolve_command.rs]: Yes, this could be clearer. Changed it to "edit".

**Pull request merged**